### DISCUSSION TITLE
Predict shap_values for new data in real-time

### DISCUSSION THREAD

**Discussion by maxreis86** [shap/shap#2539] [Q&A]
Hi dear all,

Is there a fast way to estimate shap_values for a single record in a real-time application?

For example, I have a deployed model that makes real-time prediction for a single new record.
In addition to the model prediction I would like to calculate the shap_values for each feature of that single record.
I managed to do this using the following approach:

`shap_values = saved_explainer.shap_values(df_single_record, nsamples="auto")`

But I think that this approach has two problems:

1. It is very slow. It took `13.7 s ± 558 ms` for calculate a single record. My web application can not wait that time :-(
2. The shap_values documentation says "The More samples lead to lower variance estimates of the SHAP values". I don't have good estimates of the SHAP values using a sample of single record. 

It would be perfect if I could estimate the SHAP values as fast and as accurate I performed the model prediction.

Do I have any other option?

**Comment by remo-help** [comment]
Because of the nature of how shap values are calculated, they are simply going to be slower than the actual model inference time. Different explainers and parameters can speed things up a little bit. What explainer are you using?
